packages/synchro-charts/src/components/sc-dial/sc-dial-base/sc-dial-svg.tsx

```diff
@@ -62,13 +92,91 @@ export class ScDialSvg {
       labelSize: newSize?.labelSize || LABEL_SIZE,
       unitSize: newSize?.unitSize || UNIT_SIZE,
     };
+    this.sizeChanged = true;
+  }
+
+  @Watch('percent')
+  onPercentsChanged() {
+    this.changeRing();
+  }
+
+  changeRing() {
+    const ringD: DefaultArcObject = {
+      innerRadius: DIAMETER,
+      outerRadius: DIAMETER - this.sizeConfig.dialThickness,
+      padAngle: RADIAN / 2,
+      startAngle: 0,
+      endAngle: 0,
+    };
+    const unitRadian = Math.PI * 2;
+    const angle1 = unitRadian * this.percent;
+    const angle2 = unitRadian * (1 - this.percent);
+    const currentAngle = RADIAN;
+    const endAngle1 = currentAngle + angle1;
+    const endAngle2 = endAngle1 + angle2;
+
+    const colorArc = arc()
+      .cornerRadius(CORNER_RADIUS)
+      .startAngle(currentAngle)
+      .endAngle(endAngle1);
+    const defaultArc = arc()
+      .cornerRadius(CORNER_RADIUS)
+      .startAngle(endAngle2)
+      .endAngle(endAngle1);
+
+    transition()
+      .duration(DURATION)
+      .ease(easeCubicInOut)
+      .attrTween('d', () => {
+        const endAngleColor = colorArc.endAngle()(ringD);
+        const startAngleColor = colorArc.startAngle()(ringD);
+
+        const endAngleDefault = defaultArc.endAngle()(ringD);
+        const startAngleDefault = defaultArc.startAngle()(ringD);
+
+        const interpolateColor = interpolate(
+          { startAngle: this.oldColorAngle.startAngle, endAngle: this.oldColorAngle.endAngle },
+          { startAngle: startAngleColor, endAngle: endAngleColor }
+        );
+
+        const interpolateDefault = interpolate(
+          { startAngle: this.oldDefaultAngle.startAngle, endAngle: this.oldDefaultAngle.endAngle },
+          { startAngle: startAngleColor, endAngle: endAngleColor }
+        );
+
+        this.oldColorAngle.startAngle = startAngleColor;
+        this.oldColorAngle.endAngle = endAngleColor;
+
+        this.oldDefaultAngle.startAngle = startAngleDefault;
+        this.oldDefaultAngle.endAngle = endAngleDefault;
+
+        return t => {
+          this.ColorRing = colorArc.endAngle(interpolateColor(t).endAngle)(ringD) || '';
+          this.defaultRing = defaultArc.endAngle(interpolateDefault(t).endAngle)(ringD) || '';
+          return t;
+        };
+      });
+  }
+
+  positionLabelAndIcon() {
+    if (this.showLabel) {
+      this.sizeChanged = false;
+      const DEFAULT = 0;
+      const label = this.el.querySelectorAll('text')[1];
```

Comment:
Can this be written in a more robust manner?

Reply:
be changed to 
    `const textLabel = this.el.querySelectorAll('text');
      const textLabelLength = textLabel ? textLabel.length : 0;
      const label = textLabel[textLabelLength - 1];`